dpbench/benchmarks/rodinia/pathfinder/pathfinder_numba_dpex_k.py

```diff
@@ -0,0 +1,56 @@
+# SPDX-FileCopyrightText: 2022 - 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import dpnp
+import numba_dpex
+
+
+def MIN(a, b):
+    return (a) if (a) <= (b) else (b)
+
+
+@numba_dpex.func
+def min_dpex(a, b):
+    t = a if a <= b else b
+    return t
+
+
+@numba_dpex.kernel(debug=True)
+def _pathfinder_kernel(device_src, device_dest, cols):
+    current_element = numba_dpex.get_global_id(0)
+
+    left_ind = current_element - 1 if current_element >= 1 else 0
+    right_ind = current_element + 1 if current_element < cols - 1 else cols - 1
+    up_ind = current_element
+
+    left = device_src[left_ind]
+    up = device_src[up_ind]
+    right = device_src[right_ind]
+    shortest = min_dpex(left, up)
+    shortest = min_dpex(shortest, right)
+
+    device_dest[current_element] += shortest
+
+
+def pathfinder(data, rows, cols, pyramid_height, result):
+    # create a temp list that hold first row of data as first element and empty numpy array as second element
+    device_dest = dpnp.array(data[:cols], dtype=dpnp.int64)  # first row
+    device_src = dpnp.array([0] * cols, dtype=dpnp.int64)
+
+    t = 1
+
+    while True:
+        if t >= rows:
+            break
```

Comment:
We should re-write this to be `while t < rows:`

Reply:
Re-written as `while t < rows:`